src/Common/src/Interop/Unix/System.Private.CoreLib.Native/Interop.MemReAlloc.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal unsafe partial class Sys
+    {
+        [DllImport(Interop.Libraries.CoreLibNative, EntryPoint = "CoreLibNative_MemReAlloc")]
+        internal static extern IntPtr MemReAlloc(IntPtr ptr, UIntPtr newSize);
+        
+        [DllImport(Interop.Libraries.CoreLibNative, EntryPoint = "CoreLibNative_MemGetAllocatedSize")]
+        internal static extern IntPtr MemGetAllocatedSize(IntPtr ptr);
```

Comment:
This should not be needed for xPlatforms. We do not have the capability to get the size of allocation on coreclr either and we are able to implement the required interoperability surface fine without it.

(We actually had it in coreclr pal originally, but got rid of it because it was full oof portability problems.)